### PR TITLE
Add the RELEASES.md for explaining the way to release

### PR DESCRIPTION
**Pull request by knanao**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by knanao** [comment]
We have a hack command for cherry-pick now!
https://github.com/pipe-cd/pipecd/pull/3808

File: RELEASES.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+- Push the above changes and get a review and merge.
+
+### Cut a new release
+- Before cutting a new release, wait for all jobs in GitHub Actions to pass on master branch.
+
+- Create a tagged release. The release should start with "v" and be followed by the version number.
+
+- Create a branch from the tagged release, e.g. "release-X.Y". This will be used for the stable release.
+
+## Stable release
```
Comment (nghialv) [inline RELEASES.md:45]: How about `Patch release`?

File: RELEASES.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+- Push the above changes and Create a pull request to confirm the changelog.
+  You can confirm the changelog through the reviewing comment in pull request by GitHub Actions.
+  For more information, Please see [actions-gh-release](https://github.com/pipe-cd/actions-gh-release).
+
+- Update the content in `docs/content/en/blog/releases/vX.Y.Z.md` file based on changelog results.
+  Please refer to [this example](https://github.com/pipe-cd/pipecd/pull/3792/commits/2b59f7f2a492405bf6df905b3823b417e4a10c3e).
+
+  It is recommended to commit the above changes once here.
+
+### Create Document
```
Comment (nghialv) [inline RELEASES.md:28]:
```suggestion
### Generate Document for new version
```

**Comment by knanao** [comment]
@nghialv 
Thank you.
I applied some changes, so I'm glad if you see it again.

**Comment by nghialv** [comment]
> @nghialv Thank you. I applied some changes, so I'm glad if you see it again.

Thanks. Let me check.

File: RELEASES.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
+# Release Process
+*This release process have not been fixed yet, hence this is subject to frequent change.*
```
Comment (nghialv) [inline RELEASES.md:2]: Should we remove this sentence?

Comment (knanao) [reply RELEASES.md]: Right, Thanks!

File: RELEASES.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  Please refer to [this example](https://github.com/pipe-cd/pipecd/pull/3792/commits/2b59f7f2a492405bf6df905b3823b417e4a10c3e).
+
+  It is recommended to commit the above changes once here.
+
+### Generate document for new version
+- Run the release document script
+
+  This example assumes that `vX.Y.Z` will be released:
+  ```shell
+  make release-docs version=vX.Y.Z
```
Comment (khanhtc1202) [inline RELEASES.md:31]:
```suggestion
  make gen/release-docs version=vX.Y.Z
```

File: RELEASES.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+### Generate document for new version
+- Run the release document script
+
+  This example assumes that `vX.Y.Z` will be released:
+  ```shell
+  make gen/release-docs version=vX.Y.Z
+  ````
+
+- Push the above changes and get a review and merge.
```
Comment (khanhtc1202) [inline RELEASES.md:34]:
```suggestion
- Make a pull request to `master` branch with the above changes and get reviews and merge.
```

File: RELEASES.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+### Cut a new release
+- Before cutting a new release, wait for all jobs in GitHub Actions to pass on master branch.
+
+- Create a tagged release. The release should start with "v" and be followed by the version number.
+
+- Create a branch from the tagged release, e.g. "release-X.Y". This will be used for the patch release.
+
+## Patch release
+This refers to the release of critical bug fixes.
```
Comment (khanhtc1202) [inline RELEASES.md:44]:
```suggestion
This refers to the release of critical bug fixes. \
```
nits

File: RELEASES.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+A bugfix for a functional issue (not a data loss or security issue) that only affects an alpha feature does not qualify as a critical bug fix.
+
+### Prerequisites
+- The release branch exists, e.g. "release-X.Y".
+- (optional) If you create a pull request with the cherry-pick script, `gh` is needed to be installed and ran `gh auth login`. Please refer to [cli/cli](https://github.com/cli/cli).
+
+### Fix bugs
+- Create a pull request to fix a bug on the `master` branch.
+
+- Get a review and merge.
```
Comment (khanhtc1202) [inline RELEASES.md:54]:
```suggestion
- Get reviews and merge.
```

File: RELEASES.md (hunk 6 above)
Comment (knanao) [reply RELEASES.md]: Sorry, I'm not sure about this meaning. Could you explain that difference?

Comment (knanao) [reply RELEASES.md]: Ah, I see. Thanks!

**Review by khanhtc1202 — APPROVED** [review]
Thank you 🙌

**Review by nghialv — APPROVED** [review]
Thank you.

**Pull request merged**